test/common/router/router_test.cc

```diff
@@ -4290,9 +4288,12 @@ TEST_F(RouterTest, InternalRedirectRejectedWithoutLocation) {
                     .value());
 }
 
-TEST_F(RouterTest, InternalRedirectRejectedWithBody) {
-  enableRedirects();
+TEST_F(RouterTest, InternalRedirectRejectedWithRequestBodyDisabled) {
```

Comment:
optional nit: 
InternalRedirectRejectedWithRequestBodyDisabled ->
InternalRedirectRejectedWithRequestBodyDisabledLegacy
Just to make it clear this is a legacy code path